### PR TITLE
Reserve keys starting with an underscore

### LINKED ISSUE #930: Reserve all keys beginning with `_`
The `_` prefix was used in https://github.com/WICG/attribution-reporting-api/pull/914. 

However, the change still introduced a backward incompatibility as keys prefixed with `_` are not currently reserved.

Reserving these keys would simplify future changes similar to this one.

### PR DESCRIPTION
**Pull request by agarant**
Fix #930

***
<a href="https://pr-preview.s3.amazonaws.com/agarant/attribution-reporting-api/pull/967.html" title="Last updated on Aug 23, 2023, 1:43 PM UTC (235837e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/967/871c4be...agarant:235837e.html" title="Last updated on Aug 23, 2023, 1:43 PM UTC (235837e)">Diff</a>

### CODE REVIEW

File: EVENT.md [hunk 1]
```diff
@@ -572,6 +572,11 @@ will be created.
 If the filters match for multiple event triggers, the first matching event
 trigger is used.
 
+#### Reserved keys
+
+Keys prefixed with `_` are reserved, they cannot be use other than for specified
```
Comment (dmdabbs) [inline EVENT.md:577]: use --> used

Comment (agarant) [reply EVENT.md]: thanks!

File: EVENT.md [hunk 2]
```diff
@@ -572,6 +572,11 @@ will be created.
 If the filters match for multiple event triggers, the first matching event
 trigger is used.
 
+#### Reserved keys
+
+Keys prefixed with `_` are reserved, they cannot be used other than for
```
Comment (apasel422) [inline EVENT.md:577]:
```suggestion
Keys prefixed with `_` are reserved: they cannot be used other than for
```

File: index.bs [hunk 3]
```diff
@@ -1428,6 +1428,7 @@ To <dfn>parse filter values</dfn> given a |value|:
 1. If |value| is not a [=map=], return null.
 1. Let |result| be a new [=filter map=].
 1. [=map/iterate|For each=] |filter| → |data| of |value|:
+    1. If |filter| starts with "`_`", return null.
```
Comment (apasel422) [inline index.bs:1431]: Is there a more precise algorithm we can use here, similar to https://wicg.github.io/attribution-reporting-api/#parse-an-aggregation-key-piece ?

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: index.bs (hunk 3 above)
Comment (agarant) [reply index.bs]: Updated to use: https://infra.spec.whatwg.org/#string-starts-with

**Review by apasel422 — APPROVED** [review]

**Pull request merged**